drivers/ethernet/eth_native_posix.c

```diff
@@ -0,0 +1,268 @@
+		frag = frag->frags;
+	}
+
+	net_pkt_unref(pkt);
+
+	eth_write_data(ctx->dev_fd, ctx->send, count);
+
+	return 0;
+}
+
+#if !defined(CONFIG_ETH_NATIVE_POSIX_RANDOM_MAC)
+static int mac_addr_from_str(struct eth_context *ctx, const char *src)
+{
+	unsigned int len, i;
+	char *endptr;
+
+	len = strlen(src);
+	for (i = 0; i < len; i++) {
+		if (!(src[i] >= '0' && src[i] <= '9') &&
+		    !(src[i] >= 'A' && src[i] <= 'F') &&
+		    !(src[i] >= 'a' && src[i] <= 'f') &&
+		    src[i] != ':') {
+			return -EINVAL;
+		}
+	}
+
+	memset(ctx->mac_addr, 0, sizeof(ctx->mac_addr));
+
+	for (i = 0; i < sizeof(ctx->mac_addr); i++) {
+		ctx->mac_addr[i] = strtol(src, &endptr, 16);
+		src = ++endptr;
+	}
+
+	return 0;
+}
+#endif
+
+static int eth_init(struct device *dev)
+{
+	struct eth_context *ctx = dev->driver_data;
+
+	ARG_UNUSED(ctx);
+
+	SYS_LOG_DBG("[%p] dev %p", ctx, dev);
+
+	return 0;
+}
+
+static struct net_linkaddr *eth_get_mac(struct eth_context *ctx)
+{
+	ctx->ll_addr.addr = ctx->mac_addr;
+	ctx->ll_addr.len = sizeof(ctx->mac_addr);
+
+	return &ctx->ll_addr;
+}
+
+static int read_data(struct eth_context *ctx, int fd)
+{
+	struct net_pkt *pkt;
+	struct net_buf *frag;
+	int ret;
+
+	ret = eth_read_data(fd, ctx->recv, sizeof(ctx->recv));
+	if (ret > 0) {
+		pkt = net_pkt_get_reserve_rx(0, NET_BUF_TIMEOUT);
+		if (!pkt) {
+			return -ENOMEM;
+		}
+
+		do {
+			int count = 0;
+
+			frag = net_pkt_get_frag(pkt, NET_BUF_TIMEOUT);
+			if (!frag) {
+				net_pkt_unref(pkt);
+				return -ENOMEM;
+			}
+
+			net_pkt_frag_add(pkt, frag);
+
+			net_buf_add_mem(frag, ctx->recv + count,
+					min(net_buf_tailroom(frag), ret));
+			ret -= frag->len;
+			count += frag->len;
+		} while (ret > 0);
+
+		SYS_LOG_DBG("Recv pkt %p len %d", pkt, net_pkt_get_len(pkt));
+
+		if (net_recv_data(ctx->iface, pkt) < 0) {
+			net_pkt_unref(pkt);
+		}
+	}
+
+	return 0;
+}
+
+static void eth_rx(struct eth_context *ctx)
+{
+	int ret;
+
+	NET_DBG("Starting ZETH RX thread (stack %d bytes)",
+		CONFIG_ETH_NATIVE_POSIX_RX_STACK_SIZE);
+
+	while (1) {
+		if (net_if_is_up(ctx->iface)) {
+			ret = eth_wait_data(ctx->dev_fd);
+			if (!ret) {
+				read_data(ctx, ctx->dev_fd);
+			}
+		}
+
+		k_sleep(MSEC(50));
+	}
+}
+
+static void create_rx_handler(struct eth_context *ctx)
+{
+	k_thread_create(&rx_thread_data, eth_rx_stack,
+			K_THREAD_STACK_SIZEOF(eth_rx_stack),
+			(k_thread_entry_t)eth_rx,
+			ctx, NULL, NULL, K_PRIO_COOP(14),
+			0, K_NO_WAIT);
+}
+
+static void eth_iface_init(struct net_if *iface)
+{
+	struct eth_context *ctx = net_if_get_device(iface)->driver_data;
+	struct net_linkaddr *ll_addr = eth_get_mac(ctx);
+
+	if (ctx->init_done) {
+		return;
+	}
+
+	ctx->init_done = true;
+	ctx->iface = iface;
+
+#if defined(CONFIG_ETH_NATIVE_POSIX_RANDOM_MAC)
+	/* 00-00-5E-00-53-xx Documentation RFC 7042 */
+	ctx->mac_addr[0] = 0x00;
+	ctx->mac_addr[1] = 0x00;
+	ctx->mac_addr[2] = 0x5E;
+	ctx->mac_addr[3] = 0x00;
+	ctx->mac_addr[4] = 0x53;
+	ctx->mac_addr[5] = sys_rand32_get();
```

Comment:
Just a silly note: Beware that until #6138 and #6128 are merged you will be getting exactly the same random number here all the time.

Reply:
That is fine, and if needed, user can specify unique mac address in .config file.